### DISCUSSION TITLE
API error on netlify production

### DISCUSSION THREAD

**Discussion by mrseankumar25** [prisma/prisma#16610] [Q&A]
### Bug description

I got this one once I moved my code to netlify in side try catch.

![image](https://user-images.githubusercontent.com/43328839/205455385-a941f61d-4e81-440c-a4af-4c0c0bc7646b.png)

### How to reproduce

this is my code for api:

```
import { PrismaClient } from "@prisma/client"
const prisma = new PrismaClient()

export default async function handler(req, res) {
  try {
    if (req.method === 'POST') {
      // ONLY FOR POST REQUESTS
    } else {
      const responce = await prisma.post.findMany({
        include: {
          tags: {
            include: {
              tag: {
                select: {
                  id: true,
                  name: true,
                }
              },
            },
          },
        }
      })
      res.status(200).json({ responce })
    }
  } catch (error) {
    res.status(500).json({ responce: error })
  }
}
```

### Expected behavior

this one should return all data from the "table"

### Prisma information

```
model Post {
  @@map("post")
  id String @id @default(cuid())
  tags PostsAndTags[]
  slug String @unique
  title String

  created_at DateTime @default(now())
  updated_at DateTime @updatedAt
}

model Tag {
  @@map("tag")
  id String @id @default(cuid())
  name String
  posts PostsAndTags[]

  created_at DateTime @default(now())
  updated_at DateTime @updatedAt
}

model PostsAndTag {
  @@map("postsandtag")
  id String @id @default(cuid())

  post Posts? @relation(fields: [postId], references: [id])
  postId String

  tag Tags? @relation(fields: [tagId], references: [id])
  tagId String

  created_at DateTime @default(now())
  updated_at DateTime @updatedAt
} 
```

### Environment & setup

- OS: Windows
- Database: SQLite
- Node.js version: v18.8.0
- Prisma: 4.7.1
- @prisma/client: 4.7.1

### Prisma Version

```
prisma                  : 4.7.1
@prisma/client          : 4.7.1
Current platform        : windows
Query Engine (Node-API) : libquery-engine Default_Engines_Hash (at node_modules\@prisma\engines\query_engine-windows.dll.node)
Migration Engine        : migration-engine-cli Default_Engines_Hash (at node_modules\@prisma\engines\migration-engine-windows.exe)
Introspection Engine    : introspection-core Default_Engines_Hash (at node_modules\@prisma\engines\introspection-engine-windows.exe)
Format Binary           : prisma-fmt Default_Engines_Hash (at node_modules\@prisma\engines\prisma-fmt-windows.exe)
Format Wasm             : @prisma/prisma-fmt-wasm 4.7.1-1.Default_Engines_Hash
Default Engines Hash    : Default_Engines_Hash 
Studio                  : 0.477.0
```

**Comment by mrseankumar25** [comment]
Hello there, I am curious why it's moved to discussions.

**Reply by janpio** [reply]
This is a usage problem in some form, so it is better handled in a discussion for now. If it turns out to be reproducible, we can create a bug report issue.

**Reply by mrseankumar25** [reply]
got it

**Comment by mrseankumar25** [comment]
is there anything I missed?

**Comment by janpio** [comment]
Did this happens once?
Or does it happen all the time?
If so, can you share the code to reproduce it?

**Reply by mrseankumar25** [reply]
yes it happens all the time, I added the sample code above in the "How to reproduce" section

the code belongs:
> Pages
>> Api
>>> table_api

**Reply by mrseankumar25** [reply]
By the way I'm working on nextjs app

**Reply by janpio** [reply]
Can you please create a minimal reproducible example, put it into a GitHub repository and create a new bug report issue where you provide all this information and a link to that repository? Thanks.

**Reply by mrseankumar25** [reply]
Ok I'll do it and push it on git and share here

**Reply by mrseankumar25** [reply]
Hello @janpio,

I have added Prisma in the fresh nextjs app, and I'm still facing the same issue. You can check it here: https://fdk-prisma-test.netlify.app/api/posts

this is the git repo: https://github.com/mrseankumar25/prisma-test

Thanks

**Reply by janpio** [reply]
From the logs I get when accessing `/api/posts`:

```
Dec 11, 10:07:43 PM: 5747a866 ERROR  PrismaClientInitializationError: 
Invalid `prisma.post.findMany()` invocation:

Error querying the database: unable to open database file: /var/task/node_modules/.prisma/client/./databese.db
```

It seems after whatever Next does to bundle the functions for serverless deployment, the SQLite database is not where it is supposed to be any more. That could be considered a bug (please open a bug report!), but probably no one ever tried this as this SQLite database would be read only anyway as any writes to it would not be persisted in other function calls.

**Reply by mrseankumar25** [reply]
ok, then I understand it's due to **NEXT BUNDLING PROCESS** then what's the point of giving the "SQLite" option in Prisma? if it's not allowing me to make changes in the database. If it's so, then I can use a static website instead of using Prisma. What do you think?

**Reply by janpio** [reply]
I do not know if Next.js in general is the problem, I just told you that using SQLite on Netlify does not make sense if you want to write to the database. Netlify does not work in a way where you can update a SQLite database and the changes persist.

Does the problem occur when you run your Next.js app locally? Then this might indeed be a bug worth investigating.

**Reply by mrseankumar25** [reply]
no it working fine on local

**Reply by mrseankumar25** [reply]
> I do not know if Next.js in general is the problem, I just told you that using SQLite on Netlify does not make sense if you want to write to the database. Netlify does not work in a way where you can update a SQLite database and the changes persist.
> 
> Does the problem occur when you run your Next.js app locally? Then this might indeed be a bug worth investigating.

so you mean it's not compatible with netlify, am I right?

**Reply by janpio** [reply]
Yes, writing SQLite databases will not work on serverless platforms as the file are never synced back so any writes are lost.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
